src/app/qgsattributetabledialog.h

```diff
@@ -230,12 +264,15 @@ class APP_EXPORT QgsAttributeTableDialog : public QDialog, private Ui::QgsAttrib
 
     QMenu *mFilterColumnsMenu = nullptr;
     QSignalMapper *mFilterActionMapper = nullptr;
+    QMenu *mStoredFilterExpressionMenu = nullptr;
 
     QPointer< QgsVectorLayer > mLayer = nullptr;
     QgsSearchWidgetWrapper *mCurrentSearchWidgetWrapper = nullptr;
     QStringList mVisibleFields;
     QgsAttributeEditorContext mEditorContext;
 
+    QTimer *mFilterQueryTimer;
```

Comment:
This will be leaked, no?

I think you don't need multiple of them in parallel. You just need one which is restarted on key press.

```suggestion
    QTimer mFilterQueryTimer;
```

Reply:
Also should be initialized here to nullptr if it IS required